src/Shared/UnitTests/TestEnvironment.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Shared;
+using Xunit;
+
+namespace Microsoft.Build.Engine.UnitTests
+{
+    public class TestEnvironment : IDisposable
```

Comment:
The generic nature of this is pretty neat but I think it'd be a lot nicer to read if it had helper methods for common operations.

Use patterns like `_env.WithVariant(VariantFactory.CreateFile(".proj")).File` don't read as nicely as `_env.CreateFile(".proj").Path`.